pkg/util/workqueue/prometheus/prometheus.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prometheus
+
+import (
+	"k8s.io/kubernetes/pkg/util/workqueue"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Package prometheus sets the workqueue DefaultMetricsFactory to produce
+// prometheus metrics. To use this package, you just have to import it.
+
+func init() {
+	workqueue.DefaultMetricsFactory.SetProviders(
```

Comment:
Calling a method on a type whose definition is private to another package. Not that it won't work, it just feels wrong. Like if I wanted to search godoc.org for the documentation on `SetProviders`, I wouldn't find it.